### PR TITLE
Add a controller to sync Jenkins privileges from GitHub

### PR DESCRIPTION
**Pull request by stevekuznetsov**
/cc @csrwng this is where I left off -- can't remember if I ended up being successful with the Jenkins pipeline approach or not. Today, we can reference image streams from `PodSpec` so I think we should go for a static `Job` definition in a template instead of the mess I had to do before we had that feature.

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
@csrwng updated to create a `CronJob`

File: jenkins/jobs/teams/jenkins-permissions-job.yaml [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+- apiVersion: v1
+  kind: ClusterRoleBinding
+  metadata:
+    name: "${GROUP_SYNCER_SERVICEACCOUNT}"
+  roleRef:
+    name: 'group-syncer'
+  subjects:
+  - kind: ServiceAccount
+    name: "${GROUP_SYNCER_SERVICEACCOUNT}"
+    namesapce: ???
```
Comment (stevekuznetsov) [inline jenkins/jobs/teams/jenkins-permissions-job.yaml]: @deads2k @bparees do we have any way to havethis `ClusterRoleBinding` reference the right namespace for this serviceaccount in the template today?

Comment (bparees) [inline jenkins/jobs/teams/jenkins-permissions-job.yaml]: no.  the best you can do is make a NAMESPACE parameter and make the person deploying the template put in the right value for the parameter, and then ref the parameter here.

Comment (stevekuznetsov) [inline jenkins/jobs/teams/jenkins-permissions-job.yaml]: Unfortunate. Ok, thanks

**Comment by stevekuznetsov** [comment]
@csrwng OK, this is working now. Will need to integrate with your job.

**Comment by stevekuznetsov** [comment]
I've been deploying it like this:


```sh
cat <<EOF > /tmp/sync-spec.yml
organization: origin
admins:
  - XXX
editors:
  - XXX
viewers:
  - XXX
EOF
echo -n "XXX" > /tmp/token.yml
oc new-app -f jenkins/setup/jenkins-setup-template.yaml
OAUTH_TOKEN=/tmp/token.yml SYNC_SPEC=/tmp/sync-spec.yml jenkins/jobs/teams/create-prerequisites.sh
oc new-app -f jenkins/jobs/teams/jenkins-permissions-job.yaml \
		   -p "SOURCE_URL=https://github.com/stevekuznetsov/release.git" \
		   -p "SOURCE_REF=skuznets/jenkins-github-permissions" \
		   -p "SCHEDULE=1/1 * * * *" \
		   -p "NAMESPACE=$( oc project -q )"
last_version="$( oc get bc/jenkins-team-sync -o jsonpath='{ .status.lastVersion }' )"
for (( i = 0; i < 30; i++ )); do
	if [[ "$( oc get build "jenkins-team-sync-${last_version}" -o jsonpath='{ .status.phase }' )" != "Pending" ]]; then
		break
	else
		sleep 1
	fi
done
oc logs bc/jenkins-team-sync -f
oc set image-lookup jenkins-team-sync
oc patch scheduledjob jenkins-team-sync --patch '{"spec": {"suspend":false}}'
```

**Comment by stevekuznetsov** [comment]
Jenkins secrets controller now failing with:

```
NAME      CLUSTER-IP       EXTERNAL-IP   PORT(S)   AGE
jenkins   172.30.142.251   <none>        80/TCP    18m
error: unable to list known keys: exit status 1
Traceback (most recent call last):
  File "/home/user/bin/secret-names.py", line 5, in <module>
    import oc_common
  File "/home/user/bin/oc_common.py", line 2, in <module>
    from jenkinsapi.jenkins import Jenkins
  File "/usr/lib/python2.7/site-packages/jenkinsapi/__init__.py", line 51, in <module>
    from jenkinsapi import (
  File "/usr/lib/python2.7/site-packages/jenkinsapi/api.py", line 15, in <module>
    from jenkinsapi.jenkins import Jenkins
  File "/usr/lib/python2.7/site-packages/jenkinsapi/jenkins.py", line 12, in <module>
    from jenkinsapi.jobs import Jobs
  File "/usr/lib/python2.7/site-packages/jenkinsapi/jobs.py", line 6, in <module>
    from jenkinsapi.job import Job
  File "/usr/lib/python2.7/site-packages/jenkinsapi/job.py", line 23, in <module>
    from jenkinsapi_utils.compat import to_string
ImportError: No module named jenkinsapi_utils.compat 
```

**Comment by stevekuznetsov** [comment]
Had to downgrade the version of `jenkinsapi` we were using but it all works now!!

**Pull request merged**